src/chart/generateCategoricalChart.tsx

```diff
@@ -327,6 +327,8 @@ const getAxisMapByAxes = (
 
     // we didn't create the domain from user's props above, so we need to calculate it
     if (!domain || domain.length === 0) {
+      const childDomain = child.props.domain ?? (type === 'number' ? [0, 'auto'] : undefined);
```

Comment:
It might be nice to make this a bit more verbose, to improve readability. 
This line define the default domain, given a type. 

Please do instead extract a function called 

```
getDefaultDomainByType(type) {
    return type === 'number' ? [0, 'auto'] : undefined
}

```
```suggestion
      const childDomain = child.props.domain ?? getDefaultDomainByType(type);
```

This is only required to not further complicate this largest of files with most of complexity.